### DISCUSSION TITLE
Adding a request form to app

### DISCUSSION THREAD

**Discussion by jwsmesaaz** [advplyr/audiobookshelf#2592] [Q&A] [answered]
This might be a little more technical I have a form created that my family can submit to request additional books (Instead of texting me then I forget because I'm not ever in front of a computer). The code is embedded into a webpage and works perfectly however I get text asking me what the webpage is and it is a clunky process since I'm not getting requests often. 

I'm wondering if there is a way to implement a form under the menu probably between Local Media and Settings that is "Make Request". I'm also worried about upgrades. Will I be able to upgrade the app in the future or if I do update it in the future will it erase the form? If it erases the form that is fine since I'll know how to add it back in and it will just be part of regular maintenance. If it prevents me from pulling a new version that could be an issue though.

Thanks in advance.

**Comment by nichwall** [comment]
The easiest method would probably be to make sure you fork the main server repo, then make your changes in the fork.Whenever there's an update to the upstream repo (this one) you can merge those changes into your fork.

If you only make localized changes, you can just do a fastfoward without any merge conflicts unless someone else edits the same files as you (which can happen, but then you would just need to resolve the merge conflict whenever that happens).

Once you've merged the changes from upstream, you can just recompile/run your server from source like you already would be with your changes. You can get fancier, but that'll be the simplest to implement.

**Reply by nichwall** [reply, marked as answer]
Not sure if you're the same as donquinleone on the Discord, but they may be someone to talk to as well (similar question a few days ago)

https://discord.com/channels/942908292873723984/954760207131615264/1204059887076909087
